docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,1275 @@
+k get endpointslice -n green
+NAME                  ADDRESSTYPE   PORTS   ENDPOINTS                 AGE
+service-green-ngrrc   IPv4          8080    10.244.2.5,10.244.1.8     58m
+service-green-rs2wv   IPv4          8080    104.104.2.4,104.104.1.4   58m
+---
+sh-5.2# ovn-nbctl ls-lb-list green_green.network_ovn-worker
+UUID                                    LB                  PROTO      VIP               IPs
+ab3397dd-6c4f-41bd-9239-60c73c25aaee    green_green.netw    tcp        10.96.69.71:80    104.104.1.4:8080,104.104.2.4:8080
+sh-5.2# ovn-nbctl ls-lb-list blue_blue.network_ovn-worker
+UUID                                    LB                  PROTO      VIP                 IPs
+04aa1e0b-bfec-4270-94f0-4b6d511b63a3    blue_blue.networ    tcp        10.96.210.158:80    103.103.0.5:8080,103.103.2.5:8080
+```
+
+We need the controller to also add `ab3397dd-6c4f-41bd-9239-60c73c25aaee`
+to `blue_blue.network_ovn-worker` and `04aa1e0b-bfec-4270-94f0-4b6d511b63a3`
+to `green_green.network_ovn-worker` for the blue pods to be able to
+talk to green service and vice versa. The switches will DNAT the traffic
+and the policies and routes we added in the **Pods** section will take
+care of routing the DNAT-ed traffic to the correct network.
+
+```yaml
+sh-5.2# ovn-nbctl ls-lb-add blue_blue.network_ovn-worker ab3397dd-6c4f-41bd-9239-60c73c25aaee
+sh-5.2# ovn-nbctl ls-lb-add green_green.network_ovn-worker 04aa1e0b-bfec-4270-94f0-4b6d511b63a3
+sh-5.2# ovn-nbctl ls-lb-list blue_blue.network_ovn-worker
+UUID                                    LB                  PROTO      VIP                 IPs
+04aa1e0b-bfec-4270-94f0-4b6d511b63a3    blue_blue.networ    tcp        10.96.210.158:80    103.103.0.5:8080,103.103.2.5:8080
+ab3397dd-6c4f-41bd-9239-60c73c25aaee    green_green.netw    tcp        10.96.69.71:80      104.104.1.4:8080,104.104.2.4:8080
+sh-5.2# ovn-nbctl ls-lb-list green_green.network_ovn-worker
+UUID                                    LB                  PROTO      VIP                 IPs
+04aa1e0b-bfec-4270-94f0-4b6d511b63a3    blue_blue.networ    tcp        10.96.210.158:80    103.103.0.5:8080,103.103.2.5:8080
+ab3397dd-6c4f-41bd-9239-60c73c25aaee    green_green.netw    tcp        10.96.69.71:80      104.104.1.4:8080,104.104.2.4:8080
+```
+
+Case3: `ClusterIPServiceNetwork` is set and `PodNetwork` is not set
+
+This means user is asking for partial connectivity between UDNs
+through serviceCIDRs. They don't want direct connectivity between
+all their applications. There are two potential ways to solve this:
+
+Idea1: We re-use the drop traffic between advertised UDNs logic also
+for connected UDNs. So we can do:
+
+```
+action match                                                                       priority
+------ --------------------------------------------------------------------------- --------
+pass   "(ip[4|6].src == <UDN_SUBNET> && ip[4|6].dst == <UDN_SUBNET>)"                1100
```

Comment:
is this happening on the tier 0? or won't pass interfere with some netpols?

Reply:
@dceara so this is a good point.
There were two ways I thought about:
1st: I had meant the podCIDR but to put it as "after-lb" ACL :) so pass to all connected UDN CIDRs and then drop to all other UDN Subnets below that.

2nd: what you said with pass for SVC_CIDR and then drop all other directly connected subnets 

and what I wrote was a hybrid of these two which doesn't make much sense :) I'll clear it up